### PR TITLE
fix(amplify-provider-awscloudformation): support multiprofile delete

### LINKED ISSUE #978: Cannot remove environment when checked out in another one not in the same region
Hi there !
   When checked out in an environment by example in region eu-central-1, I cannot remove another one in another region. I get an error like this:

Stack with id xxxyyyyzzz-5132437 does not exist

In same region removing one from another one is ok

### PR DESCRIPTION
**Pull request by yuth**
Amplify CLI used to configure credential at the AWS SDK root level. With multi environment support,
while deleting envs the delete request run in parallel. Since there could only be on profile set at
the SDK level, this caused deletion to fail when env used different profiles. Moved the credetianls
per service instance to support deletion of multi env

fix #978 #1323

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/lib/configuration-manager.js [hunk 1]
```diff
@@ -424,6 +424,11 @@ function persistLocalEnvConfig(context) {
 }
 
 function getCurrentConfig(context) {
+  const { envName } = context.amplify.getEnvInfo();
+  return getConfigForEnv(context, envName);
+}
+
+function getConfigForEnv(context, envName) {
```
Comment (UnleashedMind) [inline packages/amplify-provider-awscloudformation/lib/configuration-manager.js]: why are we splitting the getCurrentConfig method into two? while the envName argument is not used at all in the new method getConfigForEnv(.)

File: packages/amplify-provider-awscloudformation/lib/configuration-manager.js [hunk 2]
```diff
@@ -485,19 +490,24 @@ function removeProjectConfig(context) {
...
 
-async function loadConfiguration(context, awsClient) {
-  const projectConfigInfo = getCurrentConfig(context);
+async function loadConfiguration(context) {
+  const { envName } = context.amplify.getEnvInfo();
+  const config = await loadConfigurationForEnv(context, envName);
+  return config;
+}
+
+async function loadConfigurationForEnv(context, env) {
```
Comment (UnleashedMind) [inline packages/amplify-provider-awscloudformation/lib/configuration-manager.js]: why adding the env argument if it is not used at all, env is read from the config info file directly.

File: packages/amplify-provider-awscloudformation/lib/delete-env.js [hunk 3]
```diff
@@ -1,9 +1,10 @@
 const Cloudformation = require('../src/aws-utils/aws-cfn');
-
+const { loadConfigurationForEnv } = require('./configuration-manager');
 
 async function run(context, envName) {
-  return new Cloudformation(context)
-    .then(cfnItem => cfnItem.deleteResourceStack(envName));
+  const credentials = await loadConfigurationForEnv(context, envName);
```
Comment (UnleashedMind) [inline packages/amplify-provider-awscloudformation/lib/delete-env.js]: this does not actually return the credentials associated with the specific env.

File: packages/amplify-provider-awscloudformation/lib/configuration-manager.js (hunk 2 above)
Comment (yuth) [reply packages/amplify-provider-awscloudformation/lib/configuration-manager.js]: This was caused when rebasing from master

File: packages/amplify-provider-awscloudformation/lib/delete-env.js (hunk 3 above)
Comment (yuth) [reply packages/amplify-provider-awscloudformation/lib/delete-env.js]: We do return the config for env now

File: packages/amplify-provider-awscloudformation/lib/initializer.js [hunk 4]
```diff
@@ -58,21 +58,18 @@ async function run(context) {
 
 async function getConfiguredAwsCfnClient(context) {
```
Comment (UnleashedMind) [inline packages/amplify-provider-awscloudformation/lib/initializer.js]: If this function does not return the configured aws client anymore, but only the aws config, its name should be appropriately updated to reflect its functionality.

File: packages/amplify-provider-awscloudformation/lib/system-config-manager.js [hunk 5]
```diff
@@ -242,6 +244,16 @@ function getProfileConfig(profileName) {
   return normalizeKeys(profileConfig);
 }
 
+function loadConfigFromPath(profilePath) {
```
Comment (UnleashedMind) [inline packages/amplify-provider-awscloudformation/lib/system-config-manager.js]: This method is not needed, and should not be here, system-config-manager is only meant to  deal with system level configuration settings, such as contents in the ~/.aws/ folder and in the values in the environmental variables. 
This method is only used in the configuration-manager, in which is should be directly calling the new context.amplify.readJsonFile(.) to return aws configurations stored in the file.

File: packages/amplify-provider-awscloudformation/lib/configuration-manager.js [hunk 6]
```diff
@@ -485,19 +490,24 @@ function removeProjectConfig(context) {
...
   if (projectConfigInfo.configLevel === 'project') {
     const { config } = projectConfigInfo;
+    let awsConfig;
     if (config.useProfile) {
-      const awsConfig = await systemConfigManager.getProfiledAwsConfig(context, config.profileName);
-      awsClient.config.update(awsConfig);
+      awsConfig = await systemConfigManager.getProfiledAwsConfig(context, config.profileName);
     } else {
-      awsClient.config.loadFromPath(config.awsConfigFilePath);
+      awsConfig = systemConfigManager.loadConfigFromPath(config.awsConfigFilePath);
```
Comment (UnleashedMind) [inline packages/amplify-provider-awscloudformation/lib/configuration-manager.js]: change this line to awsConfig =  context.amplify.readJsonFile(config.awsConfigFilePath);

**Comment by yuth** [comment]
> LGTM. Did you test with profiles accross multiple accounts?

Yes. I tried multi account with multi region and it works

File: packages/amplify-provider-awscloudformation/lib/configuration-manager.js (hunk 6 above)
Comment (yuth) [reply packages/amplify-provider-awscloudformation/lib/configuration-manager.js]: we do some additional check in there. I would prefer to keep this

**Pull request merged**